File: src/Interview_exp/amzn/LoadBalancer.py
from typing import List

class Sln(object):
    def balancing_possible(nums):
        if len(a)<5:
            return False
        i=1
        j=len(a)-2
        while i!=j:
            if sum(a[:i])==sum(a[i+1:j])==sum(a[j+1:]):
                return True
            if sum(a[:i])<=sum(a[j+1:]):
                i+=1
            else:
                j-=1
        return False

print(Sln.balancing_possible([1, 3, 4, 2, 2, 2, 1, 1, 2])) # true
print(balancing_possible([1, 1, 1, 1, 1, 1])) # false
print(balancing_possible([1, 2] * 10000)) # true

File: src/Interview_exp/amzn/LoadBalancer.java
class Sln {
    static boolean solve(int[] nums) {
        int sum = 0;
        for (int n : nums)
            sum += n;
        int l = 0, r = nums.length - 1, sumL = nums[l], sumR = nums[r];
        while (l < r) {
            if (sumL == sumR) {
                if (sumL == sum - sumL * 2 - nums[l + 1] - nums[r - 1] && r - l > 1)
                    return true;
                else {
                    l++;
                    r--;
                    sumL += nums[l];
                    sumR += nums[r];
                }
            }
            else if (sumL > sumR) {
                r--;
                sumR += nums[r];
            }
            else {
                l++;
                sumL += nums[l];
            }
        }
        return false;
    }
    public static void main(String[] args) {
        int[] nums1 = {1, 3, 4, 2, 2, 2, 1, 1, 2}, nums2 = {1, 1, 1, 1, 1, 1};
        System.out.println(solve(nums1));
        System.out.println(solve(nums2));
    }
}